### ISSUE TITLE
`cannot coerce a set to a string` error when users.ldap.daemon.enable is false

### ISSUE THREAD

**Issue by a1russell** [NixOS/nixpkgs#11361]
When trying to configure LDAP, but the `users.ldap.daemon.enable` option is set to false, an error is thrown on running `nixos-rebuild`:

```
$ sudo nixos-rebuild switch --show-trace
building Nix...
building the system configuration...
error: while evaluating the attribute ‘activationScript’ of the derivation ‘nixos-15.09.597.71e29b4’ at /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/nixos/modules/system/activation/top-level.nix:102:7:
while evaluating the attribute ‘system.activationScripts.script’ at /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/nixos/modules/system/activation/activation-script.nix:58:9:
while evaluating ‘textClosureMap’ at /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/lib/strings-with-deps.nix:69:35, called from /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/nixos/modules/system/activation/activation-script.nix:79:18:
while evaluating ‘id’ at /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/lib/trivial.nix:4:8, called from undefined position:
while evaluating the attribute ‘text’ at /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/nixos/modules/system/activation/activation-script.nix:9:5:
while evaluating the attribute ‘text’ at /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/lib/strings-with-deps.nix:76:38:
cannot coerce a set to a string, at /nix/store/m3fvi30ys8pm500lb87l7q8kh11nqamj-nixos-15.09.597.71e29b4/nixos/nixpkgs/nixos/modules/config/ldap.nix:192:53
```

I am guessing this is because `${ldapConfig}` is a set, here: https://github.com/NixOS/nixpkgs/blob/release-15.09/nixos/modules/config/ldap.nix#L194

**Comment by a1russell** [comment]
Pinging @rickynils who performed last commit on that file.

**Comment by Profpatsch** [comment]
(triage) can you still reproduce this?

**Comment by a1russell** [comment]
Yes.

**Comment by obadz** [comment]
@a1russell,

I can't repro on 20f009d

```
$ cat ldap.nix 
{ config, lib, ... }:

{ users.ldap.daemon.enable = true; }

$ nix-build -I nixos-config=./ldap.nix '<nixpkgs/nixos>' -A vm
/nix/store/g9nlm0mmzmbwl2i6yw3sjs7nhih7xk3v-nixos-vm
```

Could you try to produce a minimal config file that repro's the error?

**Comment by a1russell** [comment]
Ah, I suppose there's another thing needed to recreate the bug (`bind.distinguishedName`).

```
    ldap = {
      enable = true;
      server = "";
      base = "";
      daemon.enable = false;
      bind.distinguishedName = "foo@example.com";
    };
```

**Comment by obadz** [comment]
Still works

```
$ cat ./ldap.nix 
{ config, lib, ... }:

{
  users.ldap = {
    enable = true;
    server = "";
    base = "";
    daemon.enable = false;
    bind.distinguishedName = "foo@example.com";
  };
}
$ nix-build -I nixos-config=./ldap.nix '<nixpkgs/nixos>' -A vm
/nix/store/q8lspf5643a561pc1dc3c2w9x5m9ci4l-nixos-vm
```

**Comment by globin** [comment]
> When trying to configure LDAP, but the users.ldap.daemon.enable option is set to false

I guess that's the point where it would fail?

**Comment by a1russell** [comment]
I should mention that I'm still on stable branch.

```
$ nix-build -I nixos-config=./ldap.nix '<nixpkgs/nixos>' -A vm
error: cannot coerce a set to a string, at /nix/store/dnfca8r2s9m2rf925p9r9z8rl3q35zl5-nixos-16.03.1171.9cb194c/nixos/nixpkgs/nixos/modules/config/ldap.nix:193:53
(use ‘--show-trace’ to show detailed location information)
$ cat ldap.nix 
{ config, lib, ... }:

{
  users.ldap = {
    enable = true;
    server = "";
    base = "";
    daemon.enable = false;
    bind.distinguishedName = "foo@example.com";
  };
}
$ nix-build -I nixos-config=./ldap.nix '<nixpkgs/nixos>' -A vm
error: cannot coerce a set to a string, at /nix/store/dnfca8r2s9m2rf925p9r9z8rl3q35zl5-nixos-16.03.1171.9cb194c/nixos/nixpkgs/nixos/modules/config/ldap.nix:193:53
(use ‘--show-trace’ to show detailed location information)
```

Maybe it's fixed in unstable. I can't verify that, myself, though.

**Comment by obadz** [comment]
Closed with 19ee3ba

**Comment by a1russell** [comment]
Thanks @obadz! (and @fpletz and @markus2342, of course)
